src/datachain/lib/dc.py

```diff
@@ -1992,6 +1992,7 @@ def from_records(
                 if signal_schema
                 else None
             ),
+            session=session,
```

Comment:
I think we have a few other places where create_dataset is called (and now session is not provided). Some of them might require the session object to be passed?

Reply:
The only remaining place where create_dataset is called is `enlist_source` but we don't have  option to specify session variable there. So, we will select the active session there.

Reply:
I see .. how about pull_dataset? same?

Reply:
Yes, we cannot pass session there too. The place where we have added session is the direct path from save function, where we save a new dataset and that only seem to have the ability to pass session.  That being said, if we are not passing session, we will still fetch the active context of the session.

Reply:
Checked for usage of pull_dataset and enlist_source:

- https://github.com/search?q=org%3Aiterative+pull_dataset&type=code
- https://github.com/search?q=org%3Aiterative%20enlist_source&type=code

They seem to be used in tests and CLI only, where we won't need to pass session.